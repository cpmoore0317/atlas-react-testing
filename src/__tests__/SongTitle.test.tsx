import { render } from '@testing-library/react';
import SongTitle from '../components/SongTitle';
import { expect, it, describe } from 'vitest';

describe('SongTitle Component', () => {
  it('renders song title and artist', () => {
    const { container } = render(<SongTitle title="Test Song" author="Test Artist" />);
    expect(container).toMatchSnapshot();
  });

  it('renders song title without artist', () => {
    const { container } = render(<SongTitle title="Only Title" author="" />);
    expect(container).toMatchSnapshot();
  });
});
