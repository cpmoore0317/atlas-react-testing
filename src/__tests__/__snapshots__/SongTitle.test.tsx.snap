// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Component > renders song title and artist 1`] = `
<div>
  <div
    class="text-left"
  >
    <h2
      class="text-lg font-bold"
    >
      Test Song
    </h2>
    <p
      class="text-sm text-gray-500"
    >
      Test Artist
    </p>
  </div>
</div>
`;

exports[`SongTitle Component > renders song title without artist 1`] = `
<div>
  <div
    class="text-left"
  >
    <h2
      class="text-lg font-bold"
    >
      Only Title
    </h2>
    <p
      class="text-sm text-gray-500"
    />
  </div>
</div>
`;
